src/datasources/results/ResultsDataSourceBase.ts

```diff
@@ -93,6 +79,31 @@ export abstract class ResultsDataSourceBase extends DataSourceBase<ResultsQuery>
     }
   }
 
+  async queryProducts(
+    projection?: ProductProperties[],
+  ): Promise<QueryProductResponse> {
+    try {
+      const response = await this.post<QueryProductResponse>(this.queryProductsUrl, {
+        projection,
+      });
+      return response;
+    } catch (error) {
+      throw new Error(`An error occurred while querying products: ${error}`);
+    }
+  }
+
+  async loadProducts(): Promise<QueryProductResponse> {
```

Comment:
Why are we creating a copy of the `loadProducts` method _and_ its caching mechanism to use here, instead of just leveraging `ProductUtils` that has been added in this PR?